File: eWiLL_backend/src/main/kotlin/com/wipdev/eWiLL_backend/database/tables/Diagram.kt
package com.wipdev.eWiLL_backend.database.tables

import javax.persistence.*

@Entity
@Table(name = "Diagram")
open class Diagram {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    open var id: Long? = null

    @Column(name = "name", nullable = false)
    open var name: String? = null

    @Column(name = "category_id", nullable = false)
    open var categoryId: Long? = null

    @Column(name = "ownerId", nullable = false)
    open var ownerId: Long? = null

    @Column(name = "entities", nullable = false)
    open var entities: String? = null

    @Column(name = "connections", nullable = false)
    open var connections: String? = null

}

File: modules/core/web/src/model/diagram/Diagram.ts
import type DiagramSaveType from '@/enums/DiagramSaveType'
import type DiagramConfig from './DiagramConfig'
import type Connection from './Connection'
import type Entity from './Entity'

export default interface Diagram {
  id: number
  ownerId: number
  name: string
  config: DiagramConfig
  entities: Entity[]
  connections: Connection[]
  categoryId: number
  diagramSaveType: DiagramSaveType
}
